lib/typeform_data/value_class.rb

```diff
@@ -23,6 +23,49 @@ def initialize(config, attrs)
       end
     end
 
+    # ActiveSupport defines Object#as_json in
+    # activesupport/lib/active_support/core_ext/object/json.rb which exists to separate
+    # stringification from serialization logic. as_json is not core Ruby, but we implement it here
+    # for compatibility with to_json when ActiveSupport is loaded.
+    #
+    # In addition to protection against serializing the API key, this method is needed for #to_json
+    # to work at all, since the :config object contains IO objects and ActiveSupport's #as_json
+    # method fails when trying to serialize IO objects. See
+    # https://github.com/rails/rails/issues/26132.
+    #
+    # Testing this method:
+    #
+    #   For now, we're not including ActiveSupport as a development dependency because of the risk
+    # of accidentally writing code that relies on an ActiveSupport core extension that might not
+    # be present where this gem is being used.
+    #
+    # To test this method in the console, add
+    #
+    #   spec.add_development_dependency 'activesupport'
+    #
+    # to typeform_data.gemspec and then run something like
+    #
+    #  require 'active_support'; require 'active_support/json'; \
+    #   c = TypeformData::Client.new(api_key: ...); \
+    #   at = c.all_typeforms; tf = at.reverse.find { |t| t.stats.completed > 0 }; \
+    #   tf.responses(completed: true, limit: 1).first.to_json
+    #
+    # using bin/console.
+    #
+    # One last thing: if you pass options #to_json or #as_json, Rails will use the same options for
+    # all calls in the object graph, so the following behavior is expected (if counterintuitive):
+    #
+    #    #   response.as_json(only: 'answers')
+    #   => {"answers"=>[{}, {}, {}, {}]}
+    #
+    # @param [Hash]
+    def as_json(options = {})
+      # Note: :config doesn't work here-- using the string form is mandatory.
+      return super({ except: 'config' }) if options.empty?
+
+      super(options.merge(except: ['config'] + Array(options[:except])))
```

Comment:
Thanks for the correctness change. This is and the above line amount to a a https://apidock.com/rails/Hash/deep_merge just for the `except` key, right? I'd wonder if there's a way to express it in those terms -- but probably not worth it. No action required.